src/ImmediateQRCode/IssueImmediateQRCode/Dto/Debtor.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AstrotechLabs\Itau\ImmediateQRCode\IssueImmediateQRCode\Dto;
+
+use JsonSerializable;
+use AstrotechLabs\Itau\ImmediateQRCode\IssueImmediateQRCode\ItauIssueImmediateQRCodeException;
+
+class Debtor implements JsonSerializable
+{
+    public function __construct(
+        public readonly string $name,
+        public readonly bool $isPf = false,
+        public readonly string $cpf = '',
+        public readonly string $cnpj = '',
+    ) {
+        if ($this->isPf && empty($this->cpf)) {
+            throw new ItauIssueImmediateQRCodeException(
+                code: 400,
+                key: "devedor.cpf",
+                description: "O campo CPF deve ser preenchido",
+                responsePayload: []
+            );
+        }
+
+        if (!$this->isPf && empty($this->cnpj)) {
+            throw new ItauIssueImmediateQRCodeException(
+                code: 400,
+                key: "devedor.cnpj",
+                description: "O campo CNPJ deve ser preenchido",
+                responsePayload: []
+            );
+        }
+    }
+
+    public function toArray(): array
+    {
+        return array_filter([
+            "nome" => $this->name,
+            "cpf" => preg_replace('/[^0-9]/', '', $this->cpf),
```

Comment:
@PietroCoelho Realiza esta limpeza no ato do construtor, pois neste caso do método `toArray()` você daria um return [
    'nome' => $this->name,
    'cpf' => $this->cpf,
    'cnpj' => $this->cnpj
];